cmd/cspc-operator/app/operations.go

```diff
@@ -176,6 +220,85 @@ func isRaidGroupPresentOnCSPI(group *apis.RaidGroup, cspi *apis.CStorPoolInstanc
 	return false
 }
 
+func (pc *PoolConfig) replaceExistingBlockDevice(
+	cspcRaidGroup apis.RaidGroup,
+	cspiRaidGroup *apis.RaidGroup) error {
+	cspcBlockDeviceMap := make(map[string]bool)
+	cspiBlockDeviceMap := make(map[string]bool)
+	var oldBlockDeviceName string
+	var newBlockDeviceName string
+
+	// Form CSPI Block Device Map
+	for _, bd := range cspiRaidGroup.BlockDevices {
+		cspiBlockDeviceMap[bd.BlockDeviceName] = true
+	}
+	// Form CSPC Block Device Map
+	for _, bd := range cspcRaidGroup.BlockDevices {
+		cspcBlockDeviceMap[bd.BlockDeviceName] = true
+	}
+	// Find Old Block Device Name
+	for bdName := range cspiBlockDeviceMap {
+		if !cspcBlockDeviceMap[bdName] {
+			oldBlockDeviceName = bdName
+			break
+		}
+	}
+	// Find New Block Device Name
+	for bdName := range cspcBlockDeviceMap {
+		if !cspiBlockDeviceMap[bdName] {
+			newBlockDeviceName = bdName
+			break
+		}
+	}
+
+	if oldBlockDeviceName == "" || newBlockDeviceName == "" {
+		return errors.Errorf(
+			"failed to find new block device {%s} or old block device {%s}",
+			oldBlockDeviceName,
+			newBlockDeviceName,
+		)
+	}
+
+	// Verify is that new block device is usable
+	err := pc.isBDUsable(newBlockDeviceName)
```

Comment:
as per naming func should retrun boolean value, can be fixed later as part of refactor

Reply:
We will take it as different PR